src/main/java/reposense/model/RepoConfiguration.java

```diff
@@ -25,96 +27,422 @@ public class RepoConfiguration {
+            this.repoConfiguration.isIgnoreGlobListOverriding = isIgnoreGlobListOverriding;
+            return this;
+        }
+
+        /**
+         * Updates the {@code isIgnoreCommitListOverriding} for {@code RepoConfiguration}.
+         *
+         * @param isIgnoreCommitListOverriding Checks if the list of ignored commits is overridden.
+         * @return This builder object.
+         */
+        public Builder isIgnoreCommitListOverriding(boolean isIgnoreCommitListOverriding) {
+            this.repoConfiguration.isIgnoreCommitListOverriding = isIgnoreCommitListOverriding;
+            return this;
+        }
+
+        /**
+         * Updates the {@code isFileSizeLimitOverriding} for {@code RepoConfiguration}.
+         *
+         * @param isIgnoredAuthorsListOverriding Checks if the list of ignored authors is overridden.
+         * @return This builder object.
+         */
+        public Builder isIgnoredAuthorsListOverriding(boolean isIgnoredAuthorsListOverriding) {
+            this.repoConfiguration.isIgnoredAuthorsListOverriding = isIgnoredAuthorsListOverriding;
+            return this;
+        }
+
+        /**
+         * Updates the {@code fileSizeLimit} for {@code RepoConfiguration}.
+         *
+         * @param fileSizeLimit File size limit of the repository.
+         * @return This builder object.
+         */
+        public Builder fileSizeLimit(long fileSizeLimit) {
+            this.repoConfiguration.fileSizeLimit = fileSizeLimit;
+            return this;
+        }
+
+        /**
+         * Updates the {@code isFileSizeLimitOverriding} for {@code RepoConfiguration}.
+         *
+         * @param isFileSizeLimitOverriding Checks if the file size limit is overridden.
+         * @return This builder object.
+         */
+        public Builder isFileSizeLimitOverriding(boolean isFileSizeLimitOverriding) {
+            this.repoConfiguration.isFileSizeLimitOverriding = isFileSizeLimitOverriding;
+            return this;
+        }
+
+        /**
+         * Updates the {@code isIgnoredFileAnalysisSkipped} for {@code RepoConfiguration}.
+         *
+         * @param isIgnoredFileAnalysisSkipped Checks if the analysis of ignored files is skipped.
+         * @return This builder object.
+         */
+        public Builder isIgnoredFileAnalysisSkipped(boolean isIgnoredFileAnalysisSkipped) {
+            this.repoConfiguration.isIgnoredFileAnalysisSkipped = isIgnoredFileAnalysisSkipped;
+            return this;
+        }
+
+        /**
+         * Builds the {@code RepoConfiguration} object with the necessary configurations.
+         *
+         * @return {@code RepoConfiguration}.
+         * @throws ConfigurationBuildException if there was an issue building the {@code RepoConfiguration}
+         *     object.
+         */
+        public RepoConfiguration build() {
+            if (!validate()) {
+                throw new ConfigurationBuildException();
```

Comment:
Perhaps you can add a simple test to cover this case, but looks good to me otherwise!

Reply:
Ah okay, I will work on it ASAP!